utils/config/service_configuration_test.go

```diff
@@ -333,3 +335,141 @@ func TestFlagBindingDefaults(t *testing.T) {
+	flagSet.String("host", "a host", "dummy host")
+	flagSet.String("password", "a password", "dummy password")
+	flagSet.String("user", "a user", "dummy user")
+	flagSet.String("db", "a db", "dummy db")
+	err = BindFlagToEnv(session, prefix, "TEST_DUMMY_HOST", flagSet.Lookup("host"))
+	require.NoError(t, err)
+	err = BindFlagToEnv(session, prefix, "PASSWORD", flagSet.Lookup("password"))
+	require.NoError(t, err)
+	err = BindFlagToEnv(session, prefix, "TEST_DB", flagSet.Lookup("db"))
+	require.NoError(t, err)
+	err = BindFlagToEnv(session, prefix, "DB", flagSet.Lookup("db"))
+	require.NoError(t, err)
+	err = BindFlagToEnv(session, prefix, "USER", flagSet.Lookup("user"))
+	require.NoError(t, err)
+	err = flagSet.Set("host", expectedHost)
+	require.NoError(t, err)
+	err = flagSet.Set("password", expectedPassword)
+	require.NoError(t, err)
+	err = flagSet.Set("db", expectedDB)
+	require.NoError(t, err)
+	err = LoadFromViper(session, prefix, configTest, defaults)
+	require.NoError(t, err)
+	require.NoError(t, configTest.Validate())
+
+	// Create test data
+	testValues := map[string]interface{}{
+		"TEST_DB":                 configTest.DB,
+		"TEST_DUMMY_HOST":         configTest.Host,
+		"TEST_FLAG":               configTest.Flag,
+		"TEST_HEALTHCHECK_PERIOD": configTest.HealthCheckPeriod,
+		"TEST_PASSWORD":           configTest.Password,
+		"TEST_PORT":               configTest.Port,
+		"TEST_USER":               configTest.User,
+	}
+
+	// Generate env file
+	vars, err := DetermineConfigurationEnvironmentVariables(prefix, configTest)
+	require.NoError(t, err)
+
+	// Go through generated vars and check they match the defaults
+	for key, value := range vars {
+		require.Equal(t, value, testValues[key])
+	}
+}
+
+func TestGenerateEnvFile_Nested(t *testing.T) {
+	configTest := DefaultConfiguration()
+	prefix := "test"
+
+	// Nested test values
+	/*
+		type ConfigurationTest struct {
+			TestString  string             `mapstructure:"dummy_string"`
+			TestInt     int                `mapstructure:"dummy_int"`
+			TestTime    time.Duration      `mapstructure:"dummy_time"`
+			TestConfig  DummyConfiguration `mapstructure:"dummyconfig"`  <- nested
+			TestConfig2 DummyConfiguration `mapstructure:"dummy_config"` <- nested
+		}
+
+	*/
+	testValues := map[string]interface{}{
+		"TEST_DUMMYCONFIG_TEST_DB":                  configTest.TestConfig.DB,
+		"TEST_DUMMYCONFIG_TEST_DUMMY_HOST":          configTest.TestConfig.Host,
+		"TEST_DUMMYCONFIG_TEST_FLAG":                configTest.TestConfig.Flag,
+		"TEST_DUMMYCONFIG_TEST_HEALTHCHECK_PERIOD":  configTest.TestConfig.HealthCheckPeriod,
+		"TEST_DUMMYCONFIG_TEST_PASSWORD":            configTest.TestConfig.Password,
+		"TEST_DUMMYCONFIG_TEST_PORT":                configTest.TestConfig.Port,
+		"TEST_DUMMYCONFIG_TEST_USER":                configTest.TestConfig.User,
+		"TEST_DUMMY_CONFIG_TEST_DB":                 configTest.TestConfig2.DB,
+		"TEST_DUMMY_CONFIG_TEST_DUMMY_HOST":         configTest.TestConfig2.Host,
```

Comment:
unless I am wrong the `prefix` should only appear once and the environment variable here should be : "TEST_DUMMY_CONFIG_DUMMY_HOST"